cluster.tf

```diff
@@ -61,7 +61,8 @@ resource "null_resource" "wait_for_cluster" {
   count = var.create_eks && var.manage_aws_auth ? 1 : 0
 
   depends_on = [
-    aws_eks_cluster.this[0]
+    aws_eks_cluster.this[0],
+    aws_security_group_rule.cluster_private_access,
```

Comment:
Will this work when `manage_aws_auth=false` or `cluster_endpoint_private_access=false` => when the `aws_security_group_rule.cluster_private_access` is not created ?

Reply:
I'll test `cluster_endpoint_private_access=false` now. Thank you for asking.

I have not looked at creating SGs for the cluster myself, so I do not have an answer at this time for `manage_aws_auth=false`.

Reply:
Ok. Here is the resource definition if you want to test more cases https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/cluster.tf#L48-L57

Reply:
It worked without error. `aws_security_group_rule.cluster_private_access` was created in my previous tests because we require it for security; however in my most recent test i set `cluster_endpoint_private_access = false` and `aws_security_group_rule.cluster_private_access` was not created in this run:
```
module.eks.data.helm_repository.stable: Refreshing state...
module.eks.module.eks.data.aws_iam_policy_document.cluster_assume_role_policy: Refreshing state...
module.eks.module.eks.data.aws_ami.eks_worker_windows: Refreshing state...
module.eks.module.eks.data.aws_caller_identity.current: Refreshing state...
module.eks.data.aws_subnet.subnets[1]: Refreshing state...
module.eks.data.aws_subnet.subnets[2]: Refreshing state...
module.eks.data.aws_subnet.subnets[0]: Refreshing state...
module.eks.data.aws_vpc.vpc: Refreshing state...
module.eks.module.eks.data.aws_region.current: Refreshing state...
module.eks.data.aws_caller_identity.current: Refreshing state...
module.eks.module.eks.data.aws_ami.eks_worker: Refreshing state...
module.eks.module.eks.data.aws_iam_policy_document.workers_assume_role_policy: Refreshing state...
module.eks.data.aws_region.current: Refreshing state...
module.eks.aws_ebs_encryption_by_default.default: Creating...
module.eks.module.eks.aws_security_group.cluster[0]: Creating...
module.eks.module.eks.aws_iam_role.cluster[0]: Creating...
module.eks.aws_ebs_encryption_by_default.default: Creation complete after 0s [id=terraform-20200323221013940100000003]
module.eks.module.eks.aws_iam_role.cluster[0]: Creation complete after 0s [id=gcso-k8-dev20200323221013668100000002]
module.eks.module.eks.aws_iam_role_policy_attachment.cluster_AmazonEKSServicePolicy[0]: Creating...
module.eks.module.eks.aws_iam_role_policy_attachment.cluster_AmazonEKSClusterPolicy[0]: Creating...
module.eks.module.eks.aws_iam_role_policy_attachment.cluster_AmazonEKSClusterPolicy[0]: Creation complete after 1s [id=gcso-k8-dev20200323221013668100000002-20200323221014505800000004]
module.eks.module.eks.aws_iam_role_policy_attachment.cluster_AmazonEKSServicePolicy[0]: Creation complete after 1s [id=gcso-k8-dev20200323221013668100000002-20200323221014506000000005]
module.eks.module.eks.aws_security_group.cluster[0]: Creation complete after 2s [id=sg-059c2c3600e95ab6b]
module.eks.module.eks.aws_security_group_rule.cluster_egress_internet[0]: Creating...
module.eks.module.eks.aws_eks_cluster.this[0]: Creating...
module.eks.module.eks.aws_security_group_rule.cluster_egress_internet[0]: Creation complete after 1s [id=sgrule-3304546531]
module.eks.module.eks.aws_eks_cluster.this[0]: Creation complete after 9m46s [id=gcso-k8-dev]
module.eks.data.aws_eks_cluster.cluster: Refreshing state...
module.eks.data.aws_eks_cluster_auth.cluster: Refreshing state...
module.eks.module.eks.null_resource.wait_for_cluster[0]: Creating...
module.eks.data.aws_iam_policy_document.cluster_autoscaler: Refreshing state...
module.eks.module.eks.aws_iam_role.workers[0]: Creating...
module.eks.module.eks.data.template_file.kubeconfig[0]: Refreshing state...
module.eks.module.eks.null_resource.wait_for_cluster[0]: Provisioning with 'local-exec'...
module.eks.module.eks.null_resource.wait_for_cluster[0] (local-exec): Executing: ["/bin/sh" "-c" "for i in `seq 1 60`; do wget --no-check-certificate -O - -q $ENDPOINT/healthz >/dev/null && exit 0 || true; sleep 5; done; echo TIMEOUT && exit 1"]
module.eks.module.eks.aws_security_group.workers[0]: Creating...
module.eks.aws_iam_policy.cluster_autoscaler: Creating...
module.eks.module.eks.local_file.kubeconfig[0]: Creating...
module.eks.module.eks.local_file.kubeconfig[0]: Creation complete after 0s [id=8c7cf273cfa4facd4cf0302f33eae4e182be1254]
module.eks.module.eks.aws_iam_role.workers[0]: Creation complete after 2s [id=gcso-k8-dev20200323222002411000000006]
module.eks.module.eks.aws_iam_role_policy_attachment.workers_AmazonEC2ContainerRegistryReadOnly[0]: Creating...
module.eks.module.eks.aws_iam_role_policy_attachment.workers_AmazonEKSWorkerNodePolicy[0]: Creating...
module.eks.module.eks.aws_iam_role_policy_attachment.workers_AmazonEKS_CNI_Policy[0]: Creating...
module.eks.aws_iam_policy.cluster_autoscaler: Creation complete after 2s [id=arn:aws:iam::snip:policy/cluster-autoscaler20200323222002430900000008]
module.eks.module.eks.aws_iam_role_policy_attachment.workers_additional_policies[0]: Creating...
module.eks.module.eks.aws_iam_role_policy_attachment.workers_AmazonEC2ContainerRegistryReadOnly[0]: Creation complete after 1s [id=gcso-k8-dev20200323222002411000000006-20200323222004462200000009]
module.eks.module.eks.aws_security_group.workers[0]: Creation complete after 3s [id=sg-05de39e0c1fea0514]
module.eks.module.eks.aws_security_group_rule.workers_ingress_self[0]: Creating...
module.eks.module.eks.aws_security_group_rule.workers_egress_internet[0]: Creating...
module.eks.module.eks.aws_security_group_rule.workers_ingress_cluster[0]: Creating...
module.eks.module.eks.aws_security_group_rule.cluster_https_worker_ingress[0]: Creating...
module.eks.module.eks.aws_security_group_rule.workers_ingress_cluster_https[0]: Creating...
module.eks.module.eks.aws_iam_role_policy_attachment.workers_AmazonEKSWorkerNodePolicy[0]: Creation complete after 1s [id=gcso-k8-dev20200323222002411000000006-2020032322200466440000000b]
module.eks.module.eks.aws_iam_role_policy_attachment.workers_AmazonEKS_CNI_Policy[0]: Creation complete after 1s [id=gcso-k8-dev20200323222002411000000006-2020032322200456570000000a]
module.eks.module.eks.aws_iam_role_policy_attachment.workers_additional_policies[0]: Creation complete after 2s [id=gcso-k8-dev20200323222002411000000006-2020032322200503220000000c]
module.eks.module.eks.aws_security_group_rule.workers_ingress_self[0]: Creation complete after 1s [id=sgrule-1674360278]
module.eks.module.eks.aws_security_group_rule.cluster_https_worker_ingress[0]: Creation complete after 1s [id=sgrule-1858622454]
module.eks.module.eks.aws_security_group_rule.workers_egress_internet[0]: Creation complete after 2s [id=sgrule-839947235]
module.eks.module.eks.aws_security_group_rule.workers_ingress_cluster[0]: Creation complete after 3s [id=sgrule-3505528124]
module.eks.module.eks.aws_security_group_rule.workers_ingress_cluster_https[0]: Creation complete after 4s [id=sgrule-2393933496]
module.eks.module.eks.null_resource.wait_for_cluster[0]: Creation complete after 4m22s [id=8318726415653012821]
module.eks.module.eks.kubernetes_config_map.aws_auth[0]: Creating...
module.eks.module.eks.kubernetes_config_map.aws_auth[0]: Creation complete after 0s [id=kube-system/aws-auth]
module.eks.module.eks.data.null_data_source.node_groups[0]: Refreshing state...
module.eks.module.eks.module.node_groups.random_pet.node_groups["a"]: Creating...
module.eks.module.eks.module.node_groups.random_pet.node_groups["a"]: Creation complete after 0s [id=pro-sole]
module.eks.module.eks.module.node_groups.aws_eks_node_group.workers["a"]: Creating...
module.eks.module.eks.module.node_groups.aws_eks_node_group.workers["a"]: Creation complete after 2m0s [id=gcso-k8-dev:gcso-k8-dev-a-pro-sole]
module.eks.helm_release.autoscaler: Creating...
module.eks.helm_release.autoscaler: Creation complete after 15s [id=autoscaler]

Warning: Quoted type constraints are deprecated

  on .terraform/modules/eks.metadata/variables.tf line 3, in variable "apm_id":
   3:   type        = "string"

Terraform 0.11 and earlier required type constraints to be given in quotes,
but that form is now deprecated and will be removed in a future version of
Terraform. To silence this warning, remove the quotes around "string".

(and 16 more similar warnings elsewhere)


Apply complete! Resources: 25 added, 0 changed, 0 destroyed.
module.eks.aws_ebs_encryption_by_default.default: Refreshing state... [id=terraform-20200323221013940100000003]
module.eks.module.eks.data.aws_region.current: Refreshing state...
module.eks.module.eks.data.aws_caller_identity.current: Refreshing state...
module.eks.module.eks.data.aws_ami.eks_worker: Refreshing state...
module.eks.module.eks.data.aws_iam_policy_document.cluster_assume_role_policy: Refreshing state...
module.eks.module.eks.data.aws_iam_policy_document.workers_assume_role_policy: Refreshing state...
module.eks.data.aws_vpc.vpc: Refreshing state...
module.eks.module.eks.data.aws_ami.eks_worker_windows: Refreshing state...
module.eks.data.aws_region.current: Refreshing state...
module.eks.data.aws_subnet.subnets[0]: Refreshing state...
module.eks.data.aws_subnet.subnets[1]: Refreshing state...
module.eks.data.aws_caller_identity.current: Refreshing state...
module.eks.data.aws_subnet.subnets[2]: Refreshing state...
module.eks.module.eks.aws_iam_role.cluster[0]: Refreshing state... [id=gcso-k8-dev20200323221013668100000002]
module.eks.module.eks.aws_iam_role_policy_attachment.cluster_AmazonEKSClusterPolicy[0]: Refreshing state... [id=gcso-k8-dev20200323221013668100000002-20200323221014505800000004]
module.eks.module.eks.aws_iam_role_policy_attachment.cluster_AmazonEKSServicePolicy[0]: Refreshing state... [id=gcso-k8-dev20200323221013668100000002-20200323221014506000000005]
module.eks.module.eks.aws_security_group.cluster[0]: Refreshing state... [id=sg-059c2c3600e95ab6b]
module.eks.module.eks.aws_security_group_rule.cluster_egress_internet[0]: Refreshing state... [id=sgrule-3304546531]
module.eks.module.eks.aws_eks_cluster.this[0]: Refreshing state... [id=gcso-k8-dev]
module.eks.module.eks.aws_security_group.workers[0]: Refreshing state... [id=sg-05de39e0c1fea0514]
module.eks.module.eks.data.template_file.kubeconfig[0]: Refreshing state...
module.eks.data.aws_eks_cluster.cluster: Refreshing state...
module.eks.data.aws_eks_cluster_auth.cluster: Refreshing state...
module.eks.module.eks.aws_iam_role.workers[0]: Refreshing state... [id=gcso-k8-dev20200323222002411000000006]
module.eks.data.aws_iam_policy_document.cluster_autoscaler: Refreshing state...
module.eks.module.eks.null_resource.wait_for_cluster[0]: Refreshing state... [id=8318726415653012821]
module.eks.module.eks.local_file.kubeconfig[0]: Refreshing state... [id=8c7cf273cfa4facd4cf0302f33eae4e182be1254]
module.eks.aws_iam_policy.cluster_autoscaler: Refreshing state... [id=arn:aws:iam::snip:policy/cluster-autoscaler20200323222002430900000008]
module.eks.data.helm_repository.stable: Refreshing state...
module.eks.module.eks.aws_iam_role_policy_attachment.workers_AmazonEKS_CNI_Policy[0]: Refreshing state... [id=gcso-k8-dev20200323222002411000000006-2020032322200456570000000a]
module.eks.module.eks.aws_iam_role_policy_attachment.workers_AmazonEKSWorkerNodePolicy[0]: Refreshing state... [id=gcso-k8-dev20200323222002411000000006-2020032322200466440000000b]
module.eks.module.eks.aws_iam_role_policy_attachment.workers_AmazonEC2ContainerRegistryReadOnly[0]: Refreshing state... [id=gcso-k8-dev20200323222002411000000006-20200323222004462200000009]
module.eks.module.eks.aws_security_group_rule.workers_egress_internet[0]: Refreshing state... [id=sgrule-839947235]
module.eks.module.eks.aws_security_group_rule.workers_ingress_cluster_https[0]: Refreshing state... [id=sgrule-2393933496]
module.eks.module.eks.aws_security_group_rule.workers_ingress_self[0]: Refreshing state... [id=sgrule-1674360278]
module.eks.module.eks.aws_security_group_rule.workers_ingress_cluster[0]: Refreshing state... [id=sgrule-3505528124]
module.eks.module.eks.aws_security_group_rule.cluster_https_worker_ingress[0]: Refreshing state... [id=sgrule-1858622454]
module.eks.module.eks.aws_iam_role_policy_attachment.workers_additional_policies[0]: Refreshing state... [id=gcso-k8-dev20200323222002411000000006-2020032322200503220000000c]
module.eks.module.eks.kubernetes_config_map.aws_auth[0]: Refreshing state... [id=kube-system/aws-auth]
module.eks.module.eks.data.null_data_source.node_groups[0]: Refreshing state...
module.eks.module.eks.module.node_groups.random_pet.node_groups["a"]: Refreshing state... [id=pro-sole]
module.eks.module.eks.module.node_groups.aws_eks_node_group.workers["a"]: Refreshing state... [id=gcso-k8-dev:gcso-k8-dev-a-pro-sole]
module.eks.helm_release.autoscaler: Refreshing state... [id=autoscaler]
module.eks.helm_release.autoscaler: Destroying... [id=autoscaler]
module.eks.aws_ebs_encryption_by_default.default: Destroying... [id=terraform-20200323221013940100000003]
module.eks.aws_ebs_encryption_by_default.default: Destruction complete after 0s
module.eks.helm_release.autoscaler: Destruction complete after 4s
module.eks.module.eks.aws_security_group_rule.workers_ingress_cluster[0]: Destroying... [id=sgrule-3505528124]
module.eks.module.eks.aws_security_group_rule.cluster_egress_internet[0]: Destroying... [id=sgrule-3304546531]
module.eks.module.eks.aws_iam_role_policy_attachment.workers_additional_policies[0]: Destroying... [id=gcso-k8-dev20200323222002411000000006-2020032322200503220000000c]
module.eks.module.eks.aws_security_group_rule.workers_ingress_self[0]: Destroying... [id=sgrule-1674360278]
module.eks.module.eks.aws_security_group_rule.workers_egress_internet[0]: Destroying... [id=sgrule-839947235]
module.eks.module.eks.aws_security_group_rule.workers_ingress_cluster_https[0]: Destroying... [id=sgrule-2393933496]
module.eks.module.eks.module.node_groups.aws_eks_node_group.workers["a"]: Destroying... [id=gcso-k8-dev:gcso-k8-dev-a-pro-sole]
module.eks.module.eks.aws_security_group_rule.cluster_https_worker_ingress[0]: Destroying... [id=sgrule-1858622454]
module.eks.module.eks.local_file.kubeconfig[0]: Destroying... [id=8c7cf273cfa4facd4cf0302f33eae4e182be1254]
module.eks.module.eks.local_file.kubeconfig[0]: Destruction complete after 0s
module.eks.module.eks.aws_iam_role_policy_attachment.workers_additional_policies[0]: Destruction complete after 0s
module.eks.aws_iam_policy.cluster_autoscaler: Destroying... [id=arn:aws:iam::snip:policy/cluster-autoscaler20200323222002430900000008]
module.eks.module.eks.aws_security_group_rule.cluster_egress_internet[0]: Destruction complete after 1s
module.eks.module.eks.aws_security_group_rule.workers_ingress_cluster[0]: Destruction complete after 1s
module.eks.aws_iam_policy.cluster_autoscaler: Destruction complete after 1s
module.eks.module.eks.aws_security_group_rule.cluster_https_worker_ingress[0]: Destruction complete after 1s
module.eks.module.eks.aws_security_group_rule.workers_ingress_self[0]: Destruction complete after 1s
module.eks.module.eks.aws_security_group_rule.workers_egress_internet[0]: Destruction complete after 2s
module.eks.module.eks.aws_security_group_rule.workers_ingress_cluster_https[0]: Destruction complete after 3s
module.eks.module.eks.aws_security_group.workers[0]: Destroying... [id=sg-05de39e0c1fea0514]
module.eks.module.eks.aws_security_group.workers[0]: Destruction complete after 0s
module.eks.module.eks.module.node_groups.aws_eks_node_group.workers["a"]: Destruction complete after 3m11s
module.eks.module.eks.module.node_groups.random_pet.node_groups["a"]: Destroying... [id=pro-sole]
module.eks.module.eks.module.node_groups.random_pet.node_groups["a"]: Destruction complete after 0s
module.eks.module.eks.aws_iam_role_policy_attachment.workers_AmazonEKS_CNI_Policy[0]: Destroying... [id=gcso-k8-dev20200323222002411000000006-2020032322200456570000000a]
module.eks.module.eks.kubernetes_config_map.aws_auth[0]: Destroying... [id=kube-system/aws-auth]
module.eks.module.eks.aws_iam_role_policy_attachment.workers_AmazonEKSWorkerNodePolicy[0]: Destroying... [id=gcso-k8-dev20200323222002411000000006-2020032322200466440000000b]
module.eks.module.eks.aws_iam_role_policy_attachment.workers_AmazonEC2ContainerRegistryReadOnly[0]: Destroying... [id=gcso-k8-dev20200323222002411000000006-20200323222004462200000009]
module.eks.module.eks.kubernetes_config_map.aws_auth[0]: Destruction complete after 1s
module.eks.module.eks.null_resource.wait_for_cluster[0]: Destroying... [id=8318726415653012821]
module.eks.module.eks.null_resource.wait_for_cluster[0]: Destruction complete after 0s
module.eks.module.eks.aws_iam_role_policy_attachment.workers_AmazonEKS_CNI_Policy[0]: Destruction complete after 1s
module.eks.module.eks.aws_iam_role_policy_attachment.workers_AmazonEC2ContainerRegistryReadOnly[0]: Destruction complete after 1s
module.eks.module.eks.aws_iam_role_policy_attachment.workers_AmazonEKSWorkerNodePolicy[0]: Destruction complete after 1s
module.eks.module.eks.aws_iam_role.workers[0]: Destroying... [id=gcso-k8-dev20200323222002411000000006]
module.eks.module.eks.aws_iam_role.workers[0]: Destruction complete after 1s
module.eks.module.eks.aws_eks_cluster.this[0]: Destroying... [id=gcso-k8-dev]
module.eks.module.eks.aws_eks_cluster.this[0]: Destruction complete after 9m14s
module.eks.module.eks.aws_iam_role_policy_attachment.cluster_AmazonEKSClusterPolicy[0]: Destroying... [id=gcso-k8-dev20200323221013668100000002-20200323221014505800000004]
module.eks.module.eks.aws_iam_role_policy_attachment.cluster_AmazonEKSServicePolicy[0]: Destroying... [id=gcso-k8-dev20200323221013668100000002-20200323221014506000000005]
module.eks.module.eks.aws_security_group.cluster[0]: Destroying... [id=sg-059c2c3600e95ab6b]
module.eks.module.eks.aws_iam_role_policy_attachment.cluster_AmazonEKSClusterPolicy[0]: Destruction complete after 1s
module.eks.module.eks.aws_iam_role_policy_attachment.cluster_AmazonEKSServicePolicy[0]: Destruction complete after 1s
module.eks.module.eks.aws_iam_role.cluster[0]: Destroying... [id=gcso-k8-dev20200323221013668100000002]
module.eks.module.eks.aws_security_group.cluster[0]: Destruction complete after 1s
module.eks.module.eks.aws_iam_role.cluster[0]: Destruction complete after 1s

Warning: Quoted type constraints are deprecated

  on .terraform/modules/eks.metadata/variables.tf line 3, in variable "apm_id":
   3:   type        = "string"

Terraform 0.11 and earlier required type constraints to be given in quotes,
but that form is now deprecated and will be removed in a future version of
Terraform. To silence this warning, remove the quotes around "string".

(and 16 more similar warnings elsewhere)


Destroy complete! Resources: 25 destroyed.
```